server/src-lib/Hasura/GraphQL/Transport/HTTP.hs

```diff
@@ -120,13 +112,13 @@ runHasuraGQ reqId query userInfo resolvedOp = do
   return $ encodeGQResp $ GQSuccess resp
 
 
-getMergedGQResp :: [EncJSON] -> Either String GQRespValue
+getMergedGQResp :: Traversable t=> t EncJSON -> Either String GQRespValue
```

Comment:
We need to make sure the algorithm is not slower than O(n) here where `n` is the number of remote relationships